extensions/ql-vscode/src/language-support/query-editor.ts

```diff
@@ -74,7 +74,16 @@ async function previewQueryHelp(
     const uri = Uri.file(absolutePathToMd);
     try {
       await cliServer.generateQueryHelp(pathToQhelp, absolutePathToMd);
+      // Open the raw markdown file as well as the rendered file.
+      // This will force the rendered page to refresh if there has been a change to the markdown.
+      await window.showTextDocument(uri, {
+        viewColumn: ViewColumn.Active,
+      });
       await commandManager.execute("markdown.showPreviewToSide", uri);
+      // close the editor we just opened. Users will see a brief flicker of this editor
+      // being opened, but doing so will ensure that the preview is refreshed.
+      await window.showTextDocument(uri);
+      await commandManager.execute("workbench.action.closeActiveEditor");
```

Comment:
For me, the markdown file isn't flickering open and then closing again but rather is just sitting open. The left side panel content changes from the qhelp file to the markdown file and then stays like that. It looks like:

<img width="2449" alt="Screenshot 2023-08-02 at 10 27 29" src="https://github.com/github/vscode-codeql/assets/3749000/3ab55ddc-7718-4c86-be25-bfb5210d694a">

Adjusting this to only call `showTextDocument` once instead of twice makes it work on my machine. Does this also behave the same on your machine, or is there some difference?

```suggestion
      await commandManager.execute("markdown.showPreviewToSide", uri);
      // Open the raw markdown file as well as the rendered file.
      // This will force the rendered page to refresh if there has been a change to the markdown.
      // Then close the editor we just opened. Users will see a brief flicker of this editor
      // being opened, but doing so will ensure that the preview is refreshed.
      await window.showTextDocument(uri);
      await commandManager.execute("workbench.action.closeActiveEditor");
```